### PR TITLE
Update Data Jobs Monitoring doc to add GCP Dataproc support 

### PR DESCRIPTION
**Pull request by aboitreaud**
### What does this PR do? What is the motivation?

With the aim of supporting GCP's Dataproc in DJM, this PR updates our public documentation. It creates a new Dataproc page, strongly inspired from the AWS EMR one, and adds a icon with link for it on the DJM doc landing page.

### Merge instructions

- [ ] Please merge after reviewing

### CODE REVIEW

**Review by buraizu — COMMENTED** [review]
Hi @aboitreaud, thanks for the PR. It looks like the `README.md` file still has at least one `TODO`, so I've added the `work in progress` label for now. Feel free to remove the label when the `README.md` file is complete and ready for documentation team review.

File: static/images/data_jobs/dataproc/browse-buckets.png
Comment (aboitreaud) [inline static/images/data_jobs/dataproc/browse-buckets.png:1]: I can come up with a different screenshot such as this one if you think it's more informative. 
bucket-alternative

**Comment by aboitreaud** [comment]
Thanks a lot, I should have kept it a draft. It's now ready for review so I removed the label 😃

File: content/en/data_jobs/dataproc.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+Follow these steps to enable Data Jobs Monitoring for GCP Dataproc.
+
+1. [Store your Datadog API key](#store-your-datadog-api-key-in-google-cloud-secret-manager-recommended) in GCP Secret Manager (Recommended).
+1. [Create and configure your Dataproc cluster](#create-and-configure-your-dataproc-cluster).
+1. [Specify service tagging per Spark application](#specify-service-tagging-per-spark-application).
+
+### Store your Datadog API key in Google Cloud Secret Manager (Recommended)
+1. Take note of your [Datadog API key][1].
+1. In [GCP Secret Manager][2], choose **Create secret**.
+   - Under **Name**, enter a **Secret name**. You can use `datadog_dd_api_key`.
```
Comment (paul-laffon-dd) [inline content/en/data_jobs/dataproc.md:27]: nit: `dd_api_key` name should be enough (to not repeat datadog), but it means screenshot would have to be changed

File: content/en/data_jobs/dataproc.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+   # Download and run the latest init script
+   DD_SITE=$DD_SITE DD_API_KEY=$DD_API_KEY bash -c "$(curl -L https://dd-data-jobs-monitoring-setup.s3.amazonaws.com/scripts/dataproc/dataproc_init_latest.sh) || true
+
+   ```
+
+   The script above sets the required parameters, and downloads and runs the latest init script for Data Jobs Monitoring in Dataproc. If you want to pin your script to a specific version, you can replace the file name in the URL with `dataproc_init_<version_tag>.sh`, such as `dataproc_init_1.5.0.sh` to use the specific version you want.
+
+1. On the **Customize cluster** page, find the **Initialization actions** section. Click **Browse** to bring up the **Select object** dialog.
+   {{< img src="data_jobs/dataproc/browse-buckets.png" alt="Google Cloud, Dataproc, Create a Dataproc cluster on Compute Engine, Add Initialization Action dialog. Text fields for browsing buckets" style="width:80%;" >}}
```
Comment (paul-laffon-dd) [inline content/en/data_jobs/dataproc.md:66]: I would simplify a bit this part to something like this

```
On the Customize cluster page, locate the Initialization Actions section. Enter the path where you saved the script from the previous step
```

and change the screenshot to show the `Initialization actions` part

image

File: content/en/data_jobs/dataproc.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    - link: '/data_jobs'
+      tag: 'Documentation'
+      text: 'Data Jobs Monitoring'
+---
+
+[Data Jobs Monitoring][9] gives visibility into the performance and reliability of Apache Spark applications on Google Cloud Dataproc.
+
+## Requirements
+
+[Dataproc Release 2.0.x][10] or later is required. All of Debian, Rocky Linux, and Ubuntu Dataproc standard images are supported.
```
Comment (paul-laffon-dd) [inline content/en/data_jobs/dataproc.md:13]: Add the following requirement:

This guide is for **Dataproc clusters on Compute Engine**. If you are using **Dataproc on GKE**, refer to the [Kubernetes Installation Guide](https://docs.datadoghq.com/data_jobs/kubernetes/?tab=datadogoperator) instead.

File: content/en/data_jobs/dataproc.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+### Tag spans at runtime
+
+{{% djm-runtime-tagging %}}
+
+## Further Reading
+
+{{< partial name="whats-next/whats-next.html" >}}
+
+[1]: https://app.datadoghq.com/organization-settings/api-keys
```
[the remark this answers is not in the document]
Comment (aboitreaud) [reply content/en/data_jobs/dataproc.md:99]: That's something I can't confirm, maybe @buraizu are you aware of what the good practices are for making the links work for such customers ?

File: content/en/data_jobs/dataproc.md (hunk 2 above)
Comment (aboitreaud) [reply content/en/data_jobs/dataproc.md]: Thanks, addressed !

**Review by jhgilbert — COMMENTED** [review]
I made some suggestions, and I've asked the team about whether an app.datadoghq.com link will work for customers on subdomains. I'll let you know what I find out! Feel free to ping when you want me to take another look.

File: content/en/data_jobs/dataproc.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+1. Save the following script to a GCS bucket that your Dataproc cluster can read. Take note of the path to this script.
+
+   ```shell
+   #!/bin/bash
+
+   # Set required parameter DD_SITE
+   DD_SITE={{< region-param key="dd_site" code="true" >}}
+
+   # Set required parameter DD_API_KEY with Datadog API key.
+   # The commands below assumes the API key is stored in GCP Secret Manager, with the secret name as  dd_api_key and the project <PROJECT_NAME>.
```
Comment (jhgilbert) [inline content/en/data_jobs/dataproc.md:49]:
```suggestion
   # The commands below assumes the API key is stored in GCP Secret Manager, with the secret name as dd_api_key and the project <PROJECT_NAME>.
```

File: content/en/data_jobs/dataproc.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (jhgilbert) [reply content/en/data_jobs/dataproc.md:100]: The team looked into it, and the tentative answer is that the link should work, but we aren't fully confident on it. Sorry I don't have more clarity on this one!

**Review by jhgilbert — APPROVED** [review]

**Review by fitzage — APPROVED** [review]

**Pull request merged**